airflow/providers/databricks/operators/databricks.py

```diff
@@ -720,7 +720,7 @@ class DatabricksRunNowOperator(BaseOperator):
     """
 
     # Used in airflow.models.BaseOperator
-    template_fields: Sequence[str] = ("json", "databricks_conn_id")
+    template_fields: Sequence[str] = ("json", "databricks_conn_id", "job_id")
```

Comment:
This changes is not required because this class doesn't have such of attribute and job id become a part of of json attribute which is templated.

I'm not sure is json templates even work properly in `DatabricksRunNowOperator` for XComArgs and other cases.

Reply:
I see `job_id: str | None = None,` as part of the init(). We intend to use airflow variables for job_id and this will allow for that.

Reply:
Templates fields works with class instance attributes not with constructor's arguments.

```python
class FooBar:
    def __init__(self, awesome_constructor_argument):
        self.superb_object_attribute = awesome_constructor_argument
```

Reply:
```console
FAILED tests/providers/databricks/operators/test_databricks.py::TestDatabricksRunNowOperator::test_init_with_templating - AttributeError: 'job_id' is configured as a template field but DatabricksRunNowOperator does not have this attribute.
```